includes/authentication.php

```diff
@@ -0,0 +1,81 @@
+<?php
+/**
+ * Authentication functionality
+ *
+ * @package  10up-experience
+ */
+
+namespace tenup;
+
+/**
+ * Prevent users from authenticating if they are using a weak password
+ *
+ * @param WP_User $user User object
+ * @param string  $username Username
+ * @param string  $password Password
+ *
+ * @return \WP_User|\WP_Error
+ */
+function prevent_weak_password_auth( $user, $username, $password ) {
+	$test_tlds = array( 'test', 'dev', 'local', '' );
+	$tld       = preg_replace( '#^.*\.(.*)$#', '$1', wp_parse_url( site_url(), PHP_URL_HOST ) );
+
+	if ( ! in_array( $tld, $test_tlds, true ) && in_array( strtolower( trim( $password ) ), weak_passwords(), true ) ) {
+		return new \WP_Error( 'Auth Error', sprintf( '%s <a href="%s">%s</a> %s',
```

Comment:
It probably doesn't matter so much for this plugin practically-speaking but strings can't be internationalized piece-wise like this. Only the URL should be a placeholder. I also see that @jakemgold left a comment to the same affect as I was writing this one, but I don't think "recently improved security measures" is correct either.